cmds/chmod/chmod.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2016 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// sync command in Go.
+package main
+
+import (
+	"flag"
+	"log"
+	"strconv"
+	"syscall"
+)
+
+func main() {
+
+	flag.Parse()
+	if len(flag.Args()) < 2 {
+		flag.PrintDefaults()
+		log.Fatalf("usage: chmod mode filepath")
+	}
+
+	var mode uint32
+
+	if octval, err := strconv.ParseUint(flag.Args()[0], 8, 32); err != nil {
+		log.Fatalf("Unable to decode mode. Please use a hex value. arg was %s, err was %v", flag.Args()[0], err)
+	} else if octval > 0777 {
+		log.Fatalf("Invalid octal value. Value larger than 777, was %o", octval)
+	} else {
+		mode = uint32(octval)
```

Comment:
drop the else case -- since both the above cases do a log.Fatalf, you can drop the else here and make it

```
if ... {
  ...
} else if ... {
  ...
}

mode := uint32(octval)
``` 

that also means getting rid of the `var mode uint32` declaration.